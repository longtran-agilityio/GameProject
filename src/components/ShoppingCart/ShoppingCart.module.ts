import { styled, Box, Typography } from '@mui/material'
import { Link } from 'react-router-dom'

export const StyledLink = styled(Link)({
  width: '75%',
})

export const StyledBox = styled(Box)(({ theme }) => ({
  width: theme.spacing(40),
  height: '100vh',
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'space-between',
  backgroundColor: theme.palette.primary.dark,
  padding: theme.spacing(3),
}))

export const StyledTotal = styled(Box)({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
})

export const StyledName = styled(Typography)(({ theme }) => ({
  color: theme.palette.text.secondary,
}))

export const StyledProductLists = styled(Box)(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  flex: '1 1',
  padding: theme.spacing(2, 0),
  overflow: 'auto',
}))

export const StyledProductItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  backgroundColor: theme.palette.background.paper,
  padding: theme.spacing(2),
  borderRadius: theme.spacing(1),
  color: theme.palette.text.secondary,
  marginBottom: theme.spacing(1.5),
}))

export const StyledProductItemPrice = styled(Box)(({ theme }) => ({
  display: 'flex',
}))

export const StyledPrice = styled(Typography)(({ theme }) => ({
  fontWeight: 600,
  color: theme.palette.text.secondary,
}))

export const StyledTitle = styled(Typography)(({ theme }) => ({
  fontWeight: 900,
  color: theme.palette.text.primary,
}))
