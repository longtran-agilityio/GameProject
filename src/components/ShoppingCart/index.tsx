// lib
import { Typography } from '@mui/material'
import CancelIcon from '@mui/icons-material/Cancel'
import ArrowRightAltIcon from '@mui/icons-material/ArrowRightAlt'
import { generatePath } from 'react-router-dom'
// styles
import {
  StyledBox,
  StyledTotal,
  StyledPrice,
  StyledProductLists,
  StyledProductItem,
  StyledTitle,
  StyledProductItemPrice,
  StyledName,
  StyledLink,
} from './ShoppingCart.module'
import { StyledButton } from '@webapp/components/Button/Button.module'

import { PageUrls } from '@webapp/constants/pageUrl'

import { IGame } from '@webapp/interfaces/game'

interface IShoppingCart {
  games: IGame[]
  onDelete: (game: IGame) => void
  open: boolean
  onOpen: (close: boolean) => void
  onDeleteAll: () => void
}
const ShoppingCart = ({ games, onDelete, onDeleteAll, open, onOpen }: IShoppingCart) => {
  const total = games.reduce((result, item) => result + item.prices, 0)

  const handleOpen = () => {
    onOpen(!open)
  }
  const handleDeleteAllGame = () => {
    onDeleteAll()
  }

  return (
    <StyledBox>
      <StyledTotal>
        {games.length > 0 ? (
          <>
            <StyledTitle variant='h2'>{games.length} games</StyledTitle>
            <StyledButton size='medium' disableRipple onClick={handleDeleteAllGame}>
              Clear
            </StyledButton>
          </>
        ) : (
          <StyledTitle variant='h2'>No games added</StyledTitle>
        )}
      </StyledTotal>

      <StyledProductLists>
        {games.map((item) => (
          <StyledProductItem key={item.id}>
            <StyledLink
              to={generatePath(PageUrls.GAME_DETAIL, { id: item.id.toString() })}
              onClick={handleOpen}
            >
              <StyledName>{item.name}</StyledName>
            </StyledLink>
            <StyledProductItemPrice>
              <Typography>$ {item.prices}</Typography>
              <StyledButton
                size='small'
                endIcon={<CancelIcon />}
                disableRipple
                onClick={() => onDelete(item)}
              />
            </StyledProductItemPrice>
          </StyledProductItem>
        ))}
      </StyledProductLists>

      <StyledTotal>
        <StyledPrice variant='body1'>Total: ${total.toFixed(2)}</StyledPrice>
        <StyledButton size='large' endIcon={<ArrowRightAltIcon />} disableRipple>
          Checkout
        </StyledButton>
      </StyledTotal>
    </StyledBox>
  )
}

export default ShoppingCart
