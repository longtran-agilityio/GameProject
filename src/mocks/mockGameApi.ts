import { faker } from '@faker-js/faker'

export const games = [
  {
    id: faker.datatype.number(),
    name: faker.name.firstName(),
    prices: faker.datatype.number(),
    released: faker.commerce.product(),
    background_image: faker.image.cats(),
    description_raw: faker.commerce.product(),
    genres: [
      {
        name: faker.name.firstName(),
      },
    ],
    developers: [
      {
        id: faker.datatype.number(),
        name: faker.name.firstName(),
        slug: faker.name.firstName(),
      },
    ],
    platforms: [
      {
        platform: {
          id: faker.datatype.number(),
          name: faker.name.firstName(),
          slug: faker.name.firstName(),
        },
      },
    ],
    publishers: [
      {
        id: faker.datatype.number(),
        name: faker.name.firstName(),
      },
    ],
  },
]
