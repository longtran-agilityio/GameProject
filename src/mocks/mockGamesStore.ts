import { IGame } from '@webapp/interfaces/game'

export const game: IGame[] = [
  {
    id: 3498,
    name: 'Grand Theft Auto V',
    prices: 50.12,
    released: '2013-09-17',
    background_image: 'https://media.rawg.io/media/games/456/456dea5e1c7e3cd07060c14e96612001.jpg',
    description_raw: '',
    genres: [
      {
        name: 'Action',
      },
      {
        name: 'Adventure',
      },
    ],
    developers: [
      {
        id: 3524,
        name: 'Rockstar North',
        slug: 'rockstar-north',
      },
      {
        id: 10,
        name: 'Rockstar Games',
        slug: 'rockstar-games',
      },
    ],
    platforms: [
      {
        platform: {
          id: 4,
          name: 'PC',
          slug: 'pc',
        },
      },
    ],
    publishers: [
      {
        id: 2155,
        name: 'Rockstar Games',
      },
    ],
  },
]
