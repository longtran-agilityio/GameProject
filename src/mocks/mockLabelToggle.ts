import { PageUrls } from '../constants/pageUrl'

export const labelToggle = [
  {
    label: 'Login',
    currentUrl: PageUrls.LOGIN,
  },
  {
    label: 'Register',
    currentUrl: PageUrls.REGISTER,
  },
]
