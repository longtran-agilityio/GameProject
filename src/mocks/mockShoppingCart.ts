export const mockShoppingCart = [
  {
    id: 1,
    name: 'Pokemon',
    price: 19.0,
  },
  {
    id: 2,
    name: 'Pokemon1',
    price: 19.0,
  },
  {
    id: 3,
    name: 'Pokemon2',
    price: 19.0,
  },
  {
    id: 4,
    name: 'Pokemon3',
    price: 19.0,
  },
  {
    id: 5,
    name: 'Pokemon4',
    price: 19.0,
  },
]
