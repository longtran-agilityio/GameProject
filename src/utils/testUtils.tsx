import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { theme } from '@webapp/themes/index'

const AllBrowserRouter = ({ children }: { children: React.ReactNode }) => (
  <BrowserRouter>
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  </BrowserRouter>
)

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllBrowserRouter, ...options })

export * from '@testing-library/react'
export { customRender as render }
