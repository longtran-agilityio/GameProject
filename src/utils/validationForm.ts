export const EmailValidation = {
  required: 'Email is required',
  pattern: {
    value: /^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/g,
    message: 'Invalid Email',
  },
}

export const PasswordValidation = {
  required: 'Password is required',
  minLength: { value: 8, message: 'min length is 8' },
}

export const FieldEmpty = {
  required: 'This field is required',
}
