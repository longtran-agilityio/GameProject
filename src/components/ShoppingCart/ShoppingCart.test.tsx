// libs
import { render, screen } from '@webapp/utils/testUtils'

// component
import ShoppingCart from '.'

import { game } from '@webapp/mocks/mockGamesStore'

const onClick = jest.fn()
const ShoppingCartComponent = () => {
  return render(
    <ShoppingCart
      onDelete={onClick}
      games={game}
      open={false}
      onOpen={onClick}
      onDeleteAll={onClick}
    />,
  )
}

describe('Shopping Cart', () => {
  test('should matches snapshot', () => {
    const { container } = ShoppingCartComponent()
    expect(container).toMatchSnapshot()
  })

  test('should render correctly', () => {
    ShoppingCartComponent()
    expect(screen.getByText('Clear')).toBeInTheDocument()
  })
})
