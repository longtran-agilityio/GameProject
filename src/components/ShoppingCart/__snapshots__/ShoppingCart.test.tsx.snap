// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping Cart should matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-y4ovhc"
  >
    <div
      class="MuiBox-root css-1omog99"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-18j92sb-MuiTypography-root"
      >
        1
         games
      </h2>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-17rxn03-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Clear
      </button>
    </div>
    <div
      class="MuiBox-root css-j2ham7"
    >
      <div
        class="MuiBox-root css-bdgjhp"
      >
        <a
          class="css-1hbj5v0"
          href="/game-detail/3498"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1mf8eqe-MuiTypography-root"
          >
            Grand Theft Auto V
          </p>
        </a>
        <div
          class="MuiBox-root css-gvoll6"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-14f1ohf-MuiTypography-root"
          >
            $ 
            50.12
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall css-1j30yif-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1omog99"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-wfopbl-MuiTypography-root"
      >
        Total: $
        50.12
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge css-1cjliby-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Checkout
        <span
          class="MuiButton-endIcon MuiButton-iconSizeLarge css-z4rqyd-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowRightAltIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
